src/include/sof/spinlock.h

```diff
@@ -101,11 +101,11 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 			trace_lock_value((lock)->user); \
 			panic(SOF_IPC_PANIC_DEADLOCK); /* lock not acquired */ \
 		} \
-	} while (0)
+	})
```

Comment:
GNU block expressions aren't useful when the last thing is a for loop; that is either void or not something that can be made an expression. And since this one only returns on success, there isn't need for this one either.